core/pkg/plugin/plugin_interface.go

```diff
@@ -0,0 +1,55 @@
+package plugin
+
+import (
+	"encoding/gob"
+	"net/rpc"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/model"
+)
+
+// plugin interface
+type CustomCostSource interface {
+	GetCustomCosts(req model.CustomCostRequestInterface) []model.CustomCostResponse
```

Comment:
frankly, I don't know, especially since the return value is happy with a concrete type. I don't foresee a need to have different underlying structs or anything like that, and would love to pass the struct right though. I sort of ran out of time here, thinking this is OK because it is only cast to an interface inside the calling method - all consumers of this API use structs and only structs. 

If you have any insights here, more than happy to fix this

Reply:
yup - ideas is to keep this easy to fix in the future without breaking a ton of stuff